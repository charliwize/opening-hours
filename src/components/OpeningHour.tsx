import React from 'react';
import styled from 'styled-components';
import { getDayByNumber } from '../utils/utils';

interface Props {
    item: { day: string; schedule: string[] };
}

const StyledRow = styled.div`
    display: flex;
    justify-content: space-between;
    border-bottom: 1px solid #eeeeee;
    line-height: 2.5rem;
    .day-name {
        text-transform: capitalize;
        font-weight: 600;
        font-size: 1rem;
    }
`;

const Today = styled.span`
    color: #5bcb02;
    font-size: 0.75rem;
    margin-left: 0.625rem;
    font-weight: bold;
`;

const Schedule = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: center;
    .closed-text {
        color: #a1a2a4;
    }
    div {
        align-self: self-end;
    }
`;

const WeekName = styled.div`
    display: flex;
    @media (max-width: 320px) {
        flex-direction: column;
    }
`;

const OpeningHour = ({ item }: Props) => {
    const today = new Date().getDay();
    const isCurrentDay = getDayByNumber(today) === item.day.toLocaleLowerCase();
    const { schedule, day } = item;

    return (
        <StyledRow data-testid="opening-hour">
            <WeekName>
                <div className="day-name">{day}</div>
                {isCurrentDay && <Today>TODAY</Today>}
            </WeekName>
            <Schedule>
                {schedule.map((scheduleItem, index: number) => (
                    <div className={scheduleItem === 'Closed' ? 'closed-text' : ''} key={index}>
                        {scheduleItem}
                    </div>
                ))}
            </Schedule>
        </StyledRow>
    );
};

export default OpeningHour;
