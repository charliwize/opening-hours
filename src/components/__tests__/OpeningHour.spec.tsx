import { render, screen } from '@testing-library/react';
import React from 'react';
import OpeningHour from '../OpeningHour';

const defaultProps = {
    item: {
        day: 'Saturday',
        schedule: ['9 AM - 11 AM', '4 PM - 11 PM'],
    },
};

describe('OpeningHour', () => {
    test('it renders', () => {
        const { container } = render(<OpeningHour item={defaultProps.item} />);
        expect(container).toMatchSnapshot();
    });

    test('it opening and closing times', () => {
        render(<OpeningHour item={defaultProps.item} />);
        expect(screen.getByText('9 AM - 11 AM')).toBeTruthy();
        expect(screen.getByText('4 PM - 11 PM')).toBeTruthy();
    });

    test('it renders closed text', () => {
        const props = {
            item: {
                day: 'Friday',
                schedule: ['Closed'],
            },
        };
        render(<OpeningHour item={props.item} />);
        expect(screen.getByText('Closed')).toBeTruthy();
    });
});
