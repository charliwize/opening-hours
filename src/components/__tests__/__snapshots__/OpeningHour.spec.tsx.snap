// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpeningHour it renders 1`] = `
<div>
  <div
    class="sc-bdvvtL kMUTHk"
    data-testid="opening-hour"
  >
    <div
      class="sc-hKwDye igQdvm"
    >
      <div
        class="day-name"
      >
        Saturday
      </div>
    </div>
    <div
      class="sc-dkPtRN hcoMhN"
    >
      <div
        class=""
      >
        9 AM - 11 AM
      </div>
      <div
        class=""
      >
        4 PM - 11 PM
      </div>
    </div>
  </div>
</div>
`;
